docs/synapse/devguides/power-ups.rst

```diff
@@ -0,0 +1,426 @@
+
+Storm Commands
+==============
+
+Adding Storm Commands to your Cortex via a Storm Package is a great way to extend the functionality
+of your Cortex in a CLI user-friendly way.
+
+Command Line Options
+--------------------
+
+Every **Storm** command has the ``--help`` option added automatically. This means that it is always safe to
+execute any command with ``--help`` to get a usage statement and enumerate command line arguments. The
+``desc`` field specified in the command is included in the output::
+
+    storm> acme.hello.sayhi --help
+
+    Print the hello message.
+
+    Usage: acme.hello.sayhi [options]
+
+    Options:
+
+      --help                      : Display the command usage.
+    complete. 0 nodes in 4 ms (0/sec).
+    storm>
+
+**Storm** commands may specify command line arguments using a convention which is similar (although not
+identitcal to) Python's ``argparse`` library.
+
+A more complex command declaration::
+
+  commands:
+
+    - name: acme.hello.omgopts
+      descr: |
+          This is a mult-line description containing usage examples.
+
+          // run the command with some nodes
+          inet:fqdn=acme.newp | acme.hello.omgopts vertex.link
+
+          // run the command with some command line switches
+          acme.hello.omgopts --debug --hehe haha vertex.link
+
+      cmdargs:
+
+        - - --hehe
+          - type: str
+            help: The value of the hehe optional input.
+
+        - - --debug
+          - type: bool
+            default: false
+            action: store_true
+            help: Enable debug output.
+
+        - - fqdn
+          - type: str
+            help: A mandatory / positional command line argument.
+
+A more complete example of help output::
+
+    storm> acme.hello.omgopts --help
+
+    This is a mult-line description containing usage examples.
+
+    // run the command with some nodes
+    inet:fqdn=acme.newp | acme.hello.omgopts vertex.link
+
+    // run the command with some command line switches
+    acme.hello.omgopts --debug --hehe haha vertex.link
+
+
+    Usage: acme.hello.omgopts [options] <fqdn>
+
+    Options:
+
+      --help                      : Display the command usage.
+      --hehe <hehe>               : The value of the hehe optional input.
+      --debug                     : Enable debug output.
+
+    Arguments:
+
+      <fqdn>                      : A mandatory / positional command line argument.
+    complete. 0 nodes in 6 ms (0/sec).
+
+Command line options are available within the **Storm** command by accessing the implicit
+``$cmdopts`` variable.
+
+``storm/commands/acme.hello.omgopts.storm``::
+
+    // An init {} block only runs once even if there are multiple nodes in the pipeline.
+    init {
+
+        // set global debug (once) if the user specified --debug
+        if $cmdopts.debug { $lib.debug = $lib.true } }
+
+        if ($cmdopts.hehe) { $lib.print("User Specified hehe: {hehe}", hehe=$cmdopts.hehe) }
```

Comment:
Might want to differentiate (or describe) the pattern of *not* putting certain cmdopts in init that could be set per-node

Reply:
captured as an advanced topic. non-runtsafe CLI args are rough to grok :D